### ISSUE TITLE
Downstream projects & forking the regexes

### ISSUE THREAD

**Issue by paulirish** [tobie/ua-parser#211]
**TL;DR: Downstream projects should never fork the regexes**

Any project that's parsing UAs should do it in the same standard way. There should be only one single set of regexes that are used and maintained as new UAs come out, and UA strings change.  If projects are out of sync with regexes anybody relying on them will encounter problems.

The regexes in this ua-parser project are canonical. They are maintained in YAML, which is the best fit. If downstream projects need to use them in JS, use a buildstep to pull the latest and convert to JSON.

<hr>

This also opens the issue of how projects like [Detect.js](https://github.com/darcyclarke/Detect.js) can augment the knowledge from ua-parser. Like device classification (tablet/phone/etc). 

Do we have a recommended path for people to add metadata to the results from ua-parser?

<hr> 

cc  @faisalman @darcyclarke @rafaelp @verbatim @ridjohansen  @dmolsen
